src/shelter/use-shelter.hook.ts

```diff
@@ -26,9 +27,10 @@ export const useShelter = () => {
   const createImportedAccount$ = useMemo(() => new Subject<{ privateKey: string; name: string }>(), []);
 
   useEffect(() => {
+    const hdAccounts = accounts.filter(({ type }) => type === AccountTypeEnum.HD_ACCOUNT);
     const subscriptions = [
       importWalletSubscription(importWallet$, dispatch),
-      createHdAccountSubscription(createHdAccount$, accounts, dispatch),
+      createHdAccountSubscription(createHdAccount$, hdAccounts, dispatch),
```

Comment:
do you have the correct name for the HD account?  
Imported account would not shift `Account ${number}` anymore